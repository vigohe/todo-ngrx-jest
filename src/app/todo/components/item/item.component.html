<h1 [style.text-decoration]="todo.complete ? 'line-through' : ''">
    {{todo.text}} <input type="checkbox" (change)="onChecked(todo)" [checked]="todo.complete"/> <button (click)="onRemove(todo)">X</button>
</h1>
