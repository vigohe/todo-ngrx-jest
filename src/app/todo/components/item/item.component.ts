import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { Todo } from '../../models/todo';

@Component({
  selector: 'app-item',
  templateUrl: './item.component.html',
  styleUrls: ['./item.component.css']
})
export class ItemComponent implements OnInit {

  @Input() todo: Todo;
  @Output() toggle = new EventEmitter<Todo>();
  @Output() remove = new EventEmitter<Todo>();

  constructor() { }

  ngOnInit() {
  }

  onChecked(todo: Todo) {
    this.toggle.emit(todo);
  }

  onRemove(todo: Todo) {
    this.remove.emit(todo);
  }

}
