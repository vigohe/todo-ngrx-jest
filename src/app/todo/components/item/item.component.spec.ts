import {ItemComponent} from './item.component';
import { ComponentFixture, TestBed } from '@angular/core/testing';


describe('ItemComponent', () => {
  let component: ItemComponent;
  let fixture: ComponentFixture<ItemComponent>;

  beforeEach(async() => {
    TestBed.configureTestingModule({
      imports: [],
      declarations: [ ItemComponent]
    });

    await TestBed.compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(ItemComponent);
    component = fixture.componentInstance;
    component.todo = {id: 100, text: "TODO TEXT", complete: true};

    fixture.detectChanges();
  });


  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should compile', () => {
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
