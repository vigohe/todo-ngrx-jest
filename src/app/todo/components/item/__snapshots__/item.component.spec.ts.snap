// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemComponent should compile 1`] = `
<app-item
  remove={[Function EventEmitter]}
  todo={[Function Object]}
  toggle={[Function EventEmitter]}
>
  <h1
    style="text-decoration: line-through;"
  >
    
      TODO TEXT 
    <input
      type="checkbox"
    />
     
    <button>
      X
    </button>
    
  
  </h1>
  
</app-item>
`;
